src/backend/api/Fusion.Resources.Api/Controllers/Requests/RequestsController.cs

```diff
@@ -321,8 +312,91 @@ public async Task<ActionResult<ApiContractPersonnelRequest>> ProvisionContractor
             }
         }
 
+        #region Comments
+
+        [HttpPost("/projects/{projectIdentifier}/contracts/{contractIdentifier}/resources/requests/{requestId}/comments")]
+        public async Task<ActionResult> AddRequestComment([FromRoute]ProjectIdentifier projectIdentifier, Guid contractIdentifier, Guid requestId, [FromBody] CreateRequestComment create)
+        {
+            #region Authorization
+
+            var authResult = await Request.RequireAuthorizationAsync(r =>
+            {
+                r.AlwaysAccessWhen().FullControl();
+
+                r.AnyOf(or =>
+                {
+                    or.ContractAccess(ContractRole.Any, projectIdentifier, contractIdentifier);
+                    or.BeContractorInContract(contractIdentifier);
+                });
+            });
+
+            if (authResult.Unauthorized)
+                return authResult.CreateForbiddenResponse();
+
+            #endregion
+
+            await DispatchAsync(new AddComment(requestId, create.Content, create.Origin.ToString()));
+
+            return NoContent();
+        }
+
+        [HttpPut("/projects/{projectIdentifier}/contracts/{contractIdentifier}/resources/requests/{requestId}/comments/{commentId}")]
+        public async Task<ActionResult> UpdateRequestComment(
+            [FromRoute]ProjectIdentifier projectIdentifier, Guid contractIdentifier, Guid requestId, Guid commentId, [FromBody] UpdateRequestComment update)
+        {
+            #region Authorization
+
+            var authResult = await Request.RequireAuthorizationAsync(r =>
+            {
+                r.AlwaysAccessWhen().FullControl();
+
+                r.AnyOf(or =>
+                {
+                    or.ContractAccess(ContractRole.Any, projectIdentifier, contractIdentifier);
+                    or.BeContractorInContract(contractIdentifier);
+                });
+            });
```

Comment:
Should everyone be able to update everyone's comments?

Should it be:
   AlwaysWhen -> FullControl
   r.Must().BeCommentAuthor(Guid commentId)

?

Reply:
Good point, will add auth req and handler for it.